src/components/TodoList.tsx

```diff
@@ -0,0 +1,201 @@
+import cn from 'classnames';
+import { useContext, useState } from 'react';
+import { Todo } from '../types/Todo';
+import { addTodo, deleteTodo } from '../api/todos';
+import { SetErrorContext } from '../utils/setErrorContext';
+
+interface Props {
+  todos: Todo[] | null,
+  filteringMode: string,
+  userId: number,
+  setTodos: React.Dispatch<React.SetStateAction<Todo[] | null>>,
+  todosToBeDeleted: Todo['id'][] | null,
+  setTodosToBeDeleted: React.Dispatch<React.SetStateAction<number[] | null>>,
+}
+
+let filteredTodos: Todo[] | null = [];
+
+export const TodoList: React.FC<Props> = ({
+  todos, filteringMode, userId, setTodos, todosToBeDeleted, setTodosToBeDeleted,
+}) => {
+  const [todoTitle, setTodoTitle] = useState('');
+  const [processing, setProcessing] = useState(false);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+
+  if (filteringMode !== 'all' && todos !== null) {
+    switch (filteringMode) {
+      case 'active':
+        filteredTodos = todos.filter(todo => !todo.completed);
+        break;
+      case 'completed':
+        filteredTodos = todos.filter(todo => todo.completed);
+        break;
+      default:
+    }
+  } else {
+    filteredTodos = todos;
+  }
+
+  const setError = useContext(SetErrorContext);
+
+  const handleSubmit = (event: React.KeyboardEvent<HTMLElement>) => {
+    if (event.key === 'Enter') {
+      if (todoTitle) {
```

Comment:
```suggestion
    if (event.key === 'Enter' && todoTItle) {
```

Reply:
This would require repeating the `if (event.key === 'Enter')` condition regardless for the purpose of displaying an error on trying to add a todo with an empty title